source/AI.cpp

```diff
@@ -3356,35 +3356,122 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 	}
 	else if(activeCommands.Has(Command::BOARD))
 	{
-		// Board the nearest disabled ship, focusing on hostiles before allies. Holding
-		// `Shift` results in boarding only player-owned escorts in need of assistance.
+		// Depending on the preference, board the (nearest/most expensive/mix of both) disabled ship.
+		// Pressing the board key multiple times cycles between eligible targets.
+		// It will be focusing on hostiles before allies. Holding `Shift`
+		// results in boarding only player-owned escorts in need of assistance.
 		shared_ptr<const Ship> target = ship.GetTargetShip();
-		if(!target || !CanBoard(ship, *target) || (shift && !target->IsYours()))
+		if(!target || (target && activeCommands.Has(Command::WAIT)) || !CanBoard(ship, *target)
+				|| (shift && !target->IsYours()))
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
-			bool foundAnything = false;
-			for(const shared_ptr<Ship> &other : ships)
-				if(CanBoard(ship, *other))
+			const auto targetPriority = Preferences::BoardingSetting();
+			enum class BOARDING_PRIORITY{DISTANCE, COST, MIXED};
+			const BOARDING_PRIORITY boardingPriority = targetPriority == "proximity" ? BOARDING_PRIORITY::DISTANCE :
+				targetPriority == "value" ? BOARDING_PRIORITY::COST : BOARDING_PRIORITY::MIXED;
+
+			auto strategy = [&]() noexcept -> function<double(Ship &)>
+			{
+				Point current = ship.Position();
+				switch(boardingPriority)
 				{
-					if(shift && !other->IsYours())
-						continue;
+					case BOARDING_PRIORITY::COST:
+						return [this, &ship](Ship &other) noexcept -> double
+						{
+							// Use the exact cost if the ship was scanned, otherwise use an estimation.
+							return this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+								other.Cost() : (other.ChassisCost() * 2.);
+						};
+					case BOARDING_PRIORITY::MIXED:
+						return [this, &ship, current](Ship &other) noexcept -> double
+						{
+							double cost = this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+								other.Cost() : (other.ChassisCost() * 2.);
+							return cost * cost / current.DistanceSquared(other.Position());
+						};
+					// Default to distance priorities (the default setting).
+					default:
+					case BOARDING_PRIORITY::DISTANCE:
+						return [current](Ship &other) noexcept -> double
+						{
+							return current.DistanceSquared(other.Position());
+						};
+				}
+			}();
+
+			using shipValue = pair<Ship *, double>;
+			auto boardable = vector<shipValue>{};
```

Comment:
Done, I hope I did it right
I think that processing power is more important than memory size, but I may be wrong there

Reply:
Using reserve is not about processing power, but about avoiding expensive re-allocations/re-locations when they can be avoided.
Adding an element to a vector at it's full capacity causes the complete vector to be moved, such moves are expensive and reserve() ensures that such a move only happens once (at the moment of reserve()) and not at every element that gets added.

The amount of boardable ships will typically be less than the total amount of ships, so reserve() would cause more memory to be (temporary) allocated than that is strictly needed, but I agree with warp-core that avoiding resizes during population is desirable.